### PR TITLE
Special set types

### PR DESCRIPTION
**Pull request by mforets**

### CODE REVIEW

**Comment by schillic** [comment]
Do we really want to have the `DummySet` in this library?

**Comment by mforets** [comment]
i'm not sure -- ok let's move on with EmptySet and ZeroSet here and leave that one in stand-by, i'll put it in another branch

File: src/CartesianProduct.jl [hunk 1]
```diff
@@ -59,6 +59,54 @@ function *(X::LazySet, Y::LazySet)::CartesianProduct
...
+```
+
+Right multiplication of a set by an empty set.
+
+### Input
+
+- `X` -- a convex set
+- `E` -- an empty set
+
+## Output
```
Comment (schillic) [inline src/CartesianProduct.jl]: only two "#" (also below and in `MinkowskiSum.jl`, probably copy-paste)...

**Comment by schillic** [comment]
I was thinking that, since all instances of `EmptySet` are identical, we could have one central instance and in the constructor always return this instance (aka [singleton pattern](https://en.wikipedia.org/wiki/Singleton_pattern)).

**Comment by mforets** [comment]
ok. notice that `\emptyset` returns an instance, see eg. the doctests. does it do the job?

**Comment by schillic** [comment]
> ok. notice that \emptyset returns an instance, see eg. the doctests. does it do the job?

Yes, perfect 👍

File: src/MinkowskiSum.jl [hunk 2]
```diff
@@ -77,6 +77,10 @@ Minkowski sum.
 """
 +(∅::EmptySet, X::LazySet) = EmptySet()
```
Comment (schillic) [inline src/MinkowskiSum.jl]: Return `∅` instead of creating a new instance.

Comment (schillic) [reply src/MinkowskiSum.jl]: There are four occurrences of `EmptySet()` in this file and eight occurrences in `CartesianProduct.jl`.

Comment (schillic) [reply src/MinkowskiSum.jl]: In fact the Julia compiler seems to unify all instances of this type automatically, but it is still much faster to use `∅`:
```julia
julia> @time EmptySet();
  0.000056 seconds (4 allocations: 160 bytes)
julia> @time ∅;
  0.000003 seconds (4 allocations: 160 bytes)
```

Comment (schillic) [reply src/MinkowskiSum.jl]: I am beginning to wonder if `∅` is really a constant.
```julia
julia> expand(∅)
LazySets.EmptySet()
```

Comment (schillic) [reply src/MinkowskiSum.jl]: `EmptySet()` always returns the same instance as well.
```julia
object_id(EmptySet())
0x34e36c91d4a34027
object_id(EmptySet())
0x34e36c91d4a34027
```
Then I do not understand why `∅` is faster...

**Comment by schillic** [comment]
`MethodError`: ambiguous methods for `∅+∅`, `∅*∅`, `ZeroSet(2)+ZeroSet(2)`

**Comment by schillic** [comment]
No further complaints from my side 👍

**Pull request merged**